frontend/src/metabase/search/components/InfoText/use-info-text.tsx

```diff
@@ -0,0 +1,157 @@
+import { useMemo } from "react";
+import { t } from "ttag";
+import {
+  PLUGIN_COLLECTION_COMPONENTS,
+  PLUGIN_COLLECTIONS,
+} from "metabase/plugins";
+import type { WrappedResult } from "metabase/search/types";
+import type { Collection } from "metabase-types/api";
+import { useDatabaseQuery, useTableQuery } from "metabase/common/hooks";
+import {
+  browseDatabase,
+  browseSchema,
+  collection as collectionUrl,
+  tableRowsQuery,
+} from "metabase/lib/urls";
+
+import type TableType from "metabase-lib/metadata/Table";
+
+const { CollectionAuthorityLevelIcon } = PLUGIN_COLLECTION_COMPONENTS;
+
+export type InfoTextData = {
+  link?: string | null;
+  icon?: JSX.Element | null;
+  label: string | null;
+};
+
+export const useInfoText = (result: WrappedResult): InfoTextData[] => {
```

Comment:
Here, I split the old `InfoText.jsx` file into two parts:
* `InfoText` - the typescript-ified component
* `use-info-text` - the data that's associated with this component.

I did this so I can keep consistent styling for all of the possible links that could be inserted in a search result without having to jam a component into the logic of the old components of `TableLink`, `CollectionInfoText`, `TablePath`, and so on. This helps me use one component `SearchResultLink` without worrying about the different ways that the data in each of these hooks was stored before in their separate components